src/Tizen.Uix.VoiceControlManager/Tizen.Uix.VoiceControlManager/VoiceCommandsGroup.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            _handle = handle;
+
+            _list = new List<VoiceCommand>();
+            _callback = (IntPtr vcCommand, IntPtr userData) =>
+            {
+                SafeCommandHandle cmdHandle = new SafeCommandHandle(vcCommand);
+                _list.Add(new VoiceCommand(cmdHandle));
+                return true;
+            };
+            ErrorCode error = VcCmdListForeachCommands(_handle, _callback, IntPtr.Zero);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "GetAllCommands Failed with error " + error);
+                throw ExceptionFactory.CreateException(error);
+            }
+        }
+
+        /// <summary>
+        /// The destructor of the VoiceCommandList class.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        ~VoiceCommandsGroup()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets the command list.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public IList<VoiceCommand> Commands { get => _commands; }
+
+        /// <summary>
+        /// Release any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Release any unmanaged resources used by this object.
+        /// </summary>
+        /// <param name="disposing">
+        /// If true, disposes any disposable objects. If false, does not dispose disposable objects.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+                _handle?.Dispose();
+                _handle = null;
+                _list.Clear();
+                _commands.CollectionChanged -= OnCollectionChanged;
+                _commands.Clear();
+            }
+
+            _disposed = true;
+        }
+
+        private void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add)
+            {
+                foreach (VoiceCommand item in e.NewItems)
+                    Add(item);
+            }
+            else if (e.Action == NotifyCollectionChangedAction.Remove)
+            {
+                foreach (VoiceCommand item in e.OldItems)
+                    Remove(item);
+            }
+        }
+
+        void Add(VoiceCommand command)
+        {
+            ErrorCode error = VcCmdListAdd(_handle, command._handle);
+            if (error != ErrorCode.None)
+            {
+                Log.Error(LogTag, "Add Failed with error " + error);
+                throw ExceptionFactory.CreateException(error);
+            }
+
+            _list.Add(command);
```

Comment:
you should directly add into native command list

Reply:
Removed _list variable.